src/HIE/Bios/Environment.hs

```diff
@@ -159,15 +159,21 @@ addCmdOpts cmdOpts df1 = do
 -- | Make filepaths in the given 'DynFlags' absolute.
 -- This makes the 'DynFlags' independent of the current working directory.
 makeDynFlagsAbsolute :: FilePath -> G.DynFlags -> G.DynFlags
-makeDynFlagsAbsolute work_dir df =
+makeDynFlagsAbsolute root df =
   Gap.mapOverIncludePaths makeAbs
   $ df
     { G.importPaths = map makeAbs (G.importPaths df)
     , G.packageDBFlags =
         map (Gap.overPkgDbRef makeAbs) (G.packageDBFlags df)
     }
   where
-    makeAbs = (work_dir </>)
+    makeAbs =
+#if __GLASGOW_HASKELL__ >= 903
+      case G.workingDirectory df of
```

Comment:
So, working directory is relative to the project root directory? And we always know that it is relative and not an absolute path pointing to the correct sub-directory?

Reply:
I think `(</>)` has the right semantics. I can add a comment:

```haskell
λ> "a/b" </> "/absolute"
"/absolute"
λ> "/absolute1" </> "/absolute2"
"/absolute2"
```

Reply:
It is fine this way. I just keep forgetting.